File: SecondYear/AvantikaSharma2307_AvantikaSharma_2226cseai1008_2/Week_1/pattern8.py
#User function Template for python3

class Solution:
    def printTriangle(self, N):
        for i in range(1,N+1):
            for j in range(1,N+1,1):
                if(j<=i):
                    print(j,end=" ")
                else:
                    print(" ",end=" ")
                   
            for k in range(N+1,1,-1):
                if(k>i+1):
                    print(" ",end=" ")
                else:
                    print(k-1,end=" ")
            print()

#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__':
    t = int(input())
    for _ in range(t):
        N = int(input().strip())
        ob = Solution()
        ob.printTriangle(N)
# } Driver Code Ends

File: SecondYear/chaudharyaryanpanwar_AryanPanwar_2226cs1053_2/Week_1/Pattern8.cpp
class Solution {
    public :
    void printTriangle(int n){
        //code here
         for (int i = 1 ;i <= n ; i++){
            for (int j =1 ; j<=i;j++){
                cout <<j<<" ";
            } 
            for (int j = 4*(n-i) ;j>0;j-- ){
                cout <<" ";
            }
            for (int j = i ; j>=1 ; j-- ){
                cout << j<<" ";
            }
            cout << endl;
            //16 - 12 - 8- 4
        }
    }
};